elks/tools/elf2elks/elf2elks.c

```diff
@@ -527,6 +531,60 @@ input_for_header (void)
 	"%" PRIu32 " data reloc(s).", text_n_rels, ftext_n_rels, data_n_rels);
 }
 
+static char symtab_filename[256];
+
+static void
+create_symtab (void)
+{
+    FILE *infp;
+    FILE *outfp;
+    char command[256];
+    char buf[128];
+
+    if (!symtab) return;
+
+    snprintf(symtab_filename, sizeof(symtab_filename), "%s-symXXXXXX", file_name);
```

Comment:
Since you are going to plug in the temporary file name into a command line for `popen`, and `file_name` might potentially contain unsafe characters, perhaps it might be better to exclude `file_name` from the file name template?  I.e. something like `"symXXXXXX"` rather than `"%s-symXXXXXX"`.